internal/controllers/topology/cluster/patches/engine.go

```diff
@@ -139,23 +148,82 @@ func (e *engine) Apply(ctx context.Context, blueprint *scope.ClusterBlueprint, d
 	return nil
 }
 
+// addVariablesForPatch adds variables for a given ClusterClassPatch to the items in the PatchRequest.
+func addVariablesForPatch(blueprint *scope.ClusterBlueprint, desired *scope.ClusterState, req *runtimehooksv1.GeneratePatchesRequest, definitionFrom string) error {
+	validDefinitions := definitionsForPatch(blueprint, definitionFrom)
```

Comment:
valid seems a bit misleading. the once we filter out are also valid, I guess?

Would something like definitionsForPatch fit better?